cpp/bin/torchtrtc/accuracy.cpp

```diff
@@ -19,9 +19,19 @@ bool check_rtol(const at::Tensor& diff, const std::vector<at::Tensor> inputs, fl
   return diff.abs().max().item<float>() <= threshold * maxValue;
 }
 
-bool almost_equal(const at::Tensor& a, const at::Tensor& b, float threshold) {
-  return check_rtol(a - b, {a, b}, threshold);
+bool almost_equal(const at::Tensor& a, const at::Tensor& b, float atol, float rtol) {
+  auto a_float = a.toType(at::kFloat);
+  auto b_float = b.toType(at::kFloat);
+
+  auto diff = a_float - b_float;
+  auto result = diff.abs().max().item<float>();
```

Comment:
Is this really the calculation?

Reply:
This looks like $$max(a - b) \leq a_{tol} + r_{tol} * \max|b|$$

But I am not sure how this check both rtol and atol

Reply:
The check is as per:
https://pytorch.org/docs/stable/generated/torch.allclose.html

It uses the threshold limit based on absolute tolerance limit which is directly added to calibrated reference value ( rtol * other)

Ideally, it should be true for all the values in a tensor but in this case I am checking for the maximum difference value if it is less than (adjusted + calibrated threshold) or not.

Reply:
Using this metric, please note that

almost_equal(a, b) is not the same as almost_equal(b, a)

Reply:
Can we change a and b to computed_tensor and gt_tensor?

Reply:
Changed the names for the tensors.